File: CX/ClientApp/src/servicios/ServicioSegmentos.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "segmentos";

export const ObtenerSegmentos = async () => {
    const url = `${controlador}/ObtenerSegmentos`;
    return await ProcesarDatosApi('GET', url);
}

export const AgregarSegmentos = async (data) => {
    const url = `${controlador}/insertarsegmentos`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarSegmento = async (data) => {
    const url = `${controlador}/ActualizarSegmentos`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const InactivarSegmento = async (id) => {
    const url = `${controlador}/eliminarsegmento?idSegmento=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}


export const ObtenerSegmentoPorId = async (id) => {
    const url = `${controlador}/ObtenerSegmentoPorID/${id}`;
    return await ProcesarDatosApi('GET', url);
}

File: BussinesLogic/Servicios/ServicioSegmentos.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.Segmentos;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioSegmentos : IServicioSegmentos
    {
        public readonly IRepositorioSegmentos repositorioSegmentos;

        public ServicioSegmentos(IRepositorioSegmentos repositorioSegmentos)
        {
            this.repositorioSegmentos = repositorioSegmentos;
        }
        public async Task<List<DtoSegmentos>> ObtenerSegmentos()
        {
            return await this.repositorioSegmentos.ObtenerSegmentos();
        }

        public async Task<DtoRespuestaSP> InsertarSegmentos(EntitiSegmentos entitiSegmentos)
        {
            return await this.repositorioSegmentos.InsertarSegmentos(entitiSegmentos);
        }

        public async Task<DtoRespuestaSP> ActualizarSegmentos(EntitiSegmentos entitiSegmentos)
        {
            return await this.repositorioSegmentos.ActualizarSegmentos(entitiSegmentos);
        }

        public async Task<DtoSegmentos> ObtenerSegmentoPorID(int idSegmento)
        {
            return await this.repositorioSegmentos.ObtenerSegmentosPorID(idSegmento);
        }
        public async Task<DtoRespuestaSP> EliminarSegmento(int idSegmento)
        {
            return await this.repositorioSegmentos.EliminarSegmento(idSegmento);
        }
    }
}
